vars/cloudFoundryDeploy.groovy

```diff
@@ -0,0 +1,148 @@
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationHelper
+
+import groovy.transform.Field
+
+@Field String STEP_NAME = 'cloudFoundryDeploy'
+@Field Set STEP_CONFIG_KEYS = ['cfApiEndpoint', 'cfAppName', 'cfCredentialsId', 'cfManifest', 'cfOrg', 'cfSpace', 'deployUser', 'deployTool', 'deployType', 'dockerImage', 'dockerWorkspace', 'mtaDeployParameters', 'mtaExtensionDescriptor', 'mtaPath', 'smokeTestScript', 'stashContent', 'useCAM']
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+def call(Map parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def utils = parameters.juStabUtils
+        if (utils == null) {
+            utils = new Utils()
+        }
+
+        def script = parameters.script
+        if (script == null)
+            script = [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
```

Comment:
We should discuss, whether a step should be aware of the concept of a stage

Reply:
In some usage scanarios the template is out of our control and we still want to support a stage-specific configuration without touching the Jenkinsfile.
Here we need it.

Nonetheless the order is incorrect here.
Stage configuration should overrule step configuration.
We definitely need to change this here.